### PR TITLE
7839 create route and view

### PR DESCRIPTION
**Pull request by rosiehsmith**
image

### CODE REVIEW

File: web-client/src/presenter/computeds/addressDisplayHelper.test.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+          screenMetadata: { isAssociated: false },
+        },
+      });
+
+      expect(result.showEditContacts).toEqual(false);
+    });
+  });
+
+  describe('showEditPetitionerInformation', () => {
+    it('should allow the user to edit the petitioner information if have the EDIT_PETITIONER_INFO permission', () => {
```
Comment (matthopson) [inline web-client/src/presenter/computeds/addressDisplayHelper.test.js]: tiny nit that could be addressed in a subsequent PR: just the wording looks a little off.

File: web-client/src/presenter/sequences/gotoEditPetitionerInformationInternalSequence.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { getCaseAction } from '../actions/getCaseAction';
+import { getUserPendingEmailAction } from '../actions/getUserPendingEmailAction';
+import { setCaseAction } from '../actions/setCaseAction';
+import { setCurrentPageAction } from '../actions/setCurrentPageAction';
+import { setUserPendingEmailAction } from '../actions/setUserPendingEmailAction';
+import { setupPetitionerInformationFormAction } from '../actions/setupPetitionerInformationFormAction';
+import { stopShowValidationAction } from '../actions/stopShowValidationAction';
+
+export const gotoEditPetitionerInformationInternalSequence = [
+  setCurrentPageAction('Interstitial'),
```
Comment (matthopson) [inline web-client/src/presenter/sequences/gotoEditPetitionerInformationInternalSequence.js:14]: Is there any need to do the clear alerts / errors stuff here?

File: web-client/src/views/EditPetitionerInformationInternal.jsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+          <ErrorNotification />
+
+          <h2>Edit Petitioner Information</h2>
+
+          <h3>Contact Information</h3>
+
+          <div className="blue-container margin-bottom-5">
+            <div className="usa-form-group">
+              <FormGroup errorText={validationErrors.contactPrimary?.name}>
+                <label className="usa-label" htmlFor="inCareOf">
```
Comment (matthopson) [inline web-client/src/views/EditPetitionerInformationInternal.jsx]: htmlFor looks off.

File: web-client/src/views/EditPetitionerInformationInternal.jsx [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+                    });
+                  }}
+                />
+              </FormGroup>
+            </div>
+
+            <FormGroup
+              errorText={validationErrors.contactPrimary?.additionalName}
+            >
+              <label className="usa-label" htmlFor="inCareOf">
```
Comment (matthopson) [inline web-client/src/views/EditPetitionerInformationInternal.jsx]: Same here: htmlFor

File: web-client/src/router.js [hunk 5]
```diff
@@ -212,6 +212,19 @@ const router = {
       }),
     );
 
+    registerRoute(
+      '/case-detail/*/edit-petitioner-information/*',
+      ifHasAccess((docketNumber, contactId) => {
+        setPageTitle(`Docket ${docketNumber}`);
```
Comment (matthopson) [inline web-client/src/router.js]: Do we want a more descriptive page title here? Maybe doesn't matter either way.

File: web-client/src/views/EditPetitionerInformationInternal.jsx [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+                <span>
+                  Additional name <span className="usa-hint">(optional)</span>
+                </span>
+                <p className="usa-hint">
+                  A representative of the taxpayer or petitioner (In Care Of,
```
Comment (matthopson) [inline web-client/src/views/EditPetitionerInformationInternal.jsx:85]: ~How is this different from the contact `title`?~ NVM. thanks for clearing this up!

File: web-client/src/views/EditPetitionerInformationInternal.jsx [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+                New email address
+              </label>
+              <input
+                autoCapitalize="none"
+                className="usa-input"
+                id="email"
+                name="email"
+                type="text"
+                value={form.email || ''}
+                // onBlur={() => validateChangeLoginAndServiceEmailSequence()}
```
Comment (matthopson) [inline web-client/src/views/EditPetitionerInformationInternal.jsx:185]: Intentionally commented out?

File: web-client/src/views/EditPetitionerInformationInternal.jsx [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+                Re-enter new email address
+              </label>
+              <input
+                autoCapitalize="none"
+                className="usa-input"
+                id="confirm-email"
+                name="confirmEmail"
+                type="text"
+                value={form.confirmEmail || ''}
+                // onBlur={() => validateChangeLoginAndServiceEmailSequence()}
```
Comment (matthopson) [inline web-client/src/views/EditPetitionerInformationInternal.jsx:205]: Same here (I assume it's intentional, just making sure)

**Review by matthopson — APPROVED** [review]
Really a bunch of nits that could either be ignored, or addressed in another PR (assuming you've got more coming :P ), if you want to save the build time and merge as-is.

File: web-client/src/views/EditPetitionerInformationInternal.jsx (hunk 7 above)
Comment (rachelschneiderman) [reply web-client/src/views/EditPetitionerInformationInternal.jsx:185]: Yes! We are working on populating the page with the correct information now in a different branch :)

File: web-client/src/views/EditPetitionerInformationInternal.jsx (hunk 8 above)
Comment (rachelschneiderman) [reply web-client/src/views/EditPetitionerInformationInternal.jsx:205]: Yep! Thanks for checking!

**Pull request merged**